### ISSUE TITLE
I want the blocks to be usable.

### ISSUE THREAD

**Issue by vinnyg0621** [Aeltumn/Chisels-and-Bits-2#20]
I want the blocks to be usable, like something you can sit on, it can be a door, a ladder, or a storage block.

**Comment by Aeltumn** [comment]
I can ramble for absolute ages about adding "functional blocks" with which I'm talking about anything that makes C&B blocks anything more than decoration. E.g. storage blocks, custom light level, custom slipperiness, fake ladders, chiselled workbenches or even functional or animated moving blocks. (like pistons or animated doors)

There are lots of possible functional blocks that can be made but my main problem is the ability to make a complex functional block simple to use, understand and set up. Take a storage block, how many slots should there be? Which bits should active opening the container, all of a certain type, a certain set of bits, the entire block? How would you set up which bits would cause the container to be opened?

Many of the functional blocks would require tons and tons of GUIs which would quickly destroy the ability for new users to understand the entire mod. Ideally interacting with the mod should be simple and complexity should be a logical result of combining the tools provided so players get the satisfaction of figuring out cool ways to apply the tools on their own. That's hard to design but even harder the more complicated the features to explain.

Basically, tools should be clear from just their name or a simple one-line description and have clear basic uses. But upon closer inspection players should discover interesting applications of the various ways to use the tools which bring complexity. Most functional blocks already require complicated tools to even get a basic functional block working which is exactly opposite of my design goal.

tl;dr no, they are too complex for C&B, maybe in the future

**Comment by vinnyg0621** [comment]
Yes! That's what I was talking about! 😄

**Comment by Aeltumn** [comment]
I've updated the title to be more descriptive of your request if you don't mind.

**Comment by vinnyg0621** [comment]
Ok. Thank you for updating it. :)

**Comment by LordMZTE** [comment]
I think blocks inheriting properties of the bits they are made of is a good idea (like light and slipperiness) as it is intuitive and adds to the immersion. however, about things that can move or act as storage, I do agree that they wouldn't fit the mod. besides that, little tiles has that feature already

**Comment by lilmrmagoo** [comment]
a way to do it could be instead of making the chiselled block be the thing what about making it more of like a skin where you have a table and you put in a chest and a chiselled block and then get a chiselled block that works just like a chest. this I think would be pretty intuitive for me players and wouldn't require lots of guis another way you could do is a linking to where you click on one block say a chest then onto the chiselled block and it would link it so if you interacted with the chiselled block it would function like the linked block.
